Archipelago.MultiClient.Net.Analyzers/Archipelago.MultiClient.Net.Analyzers/Fixes/ItemFlagsInSwitchFixes.cs

```diff
@@ -0,0 +1,129 @@
+using System.Threading.Tasks;
+
+namespace Archipelago.MultiClient.Net.Analyzers.Fixes
+{
+    [ExportCodeFixProvider(LanguageNames.CSharp), Shared]
+    public class ItemFlagsInSwitchFixes : CodeFixProvider
+    {
+        public const string FixKeyConvertItemFlagsSwitch = "ConvertItemFlagsSwitch";
+
+        public override ImmutableArray<string> FixableDiagnosticIds => ImmutableArray.Create(
+            Constants.DiagnosticPrefix + "003"
+        );
+
+        public override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            SyntaxNode? root = await context.Document.GetSyntaxRootAsync(context.CancellationToken)
+                .ConfigureAwait(false);
+            if (root == null)
+            {
+                return;
+            }
+
+            Diagnostic diagnostic = context.Diagnostics.First();
+            TextSpan span = diagnostic.Location.SourceSpan;
+            CaseSwitchLabelSyntax? caseLabel = root.FindToken(span.Start).Parent?
+                .FirstAncestorOrSelf<CaseSwitchLabelSyntax>();
+            if (caseLabel == null)
+            {
+                return;
+            }
+
+            context.RegisterCodeFix(
+                CodeAction.Create(
+                    title: "Convert case to use HasFlag",
+                    createChangedDocument: c => ConvertCaseToHasFlag(
+                        document: context.Document,
+                        caseLabel: caseLabel,
+                        cancellationToken: c
+                    ),
+                    equivalenceKey: FixKeyConvertItemFlagsSwitch
+                ),
+                diagnostic
+            );
+        }
+
+        private async Task<Document> ConvertCaseToHasFlag(
+            Document document,
+            CaseSwitchLabelSyntax caseLabel,
+            CancellationToken cancellationToken)
+        {
+            DocumentEditor editor = await DocumentEditor.CreateAsync(document, cancellationToken);
+            SwitchSectionSyntax switchSection = (SwitchSectionSyntax)caseLabel.Parent!;
+            SwitchStatementSyntax switchStatement = (SwitchStatementSyntax)switchSection.Parent!;
+            ExpressionSyntax switchExpression = switchStatement.Expression;
+            
+            string variableName = GetPatternMatchingVariableName(editor, switchStatement);
+
+            // Generate the pattern matching case label
+            CasePatternSwitchLabelSyntax newCaseLabel = GeneratePatternMatchingCaseLabel(caseLabel, switchExpression, variableName);
+
+            // Replace the old case label with the new case label
+            editor.ReplaceNode(caseLabel, newCaseLabel);
+
+            Document newDoc = editor.GetChangedDocument();
+            return newDoc;
+        }
+
+        private static string GetPatternMatchingVariableName(DocumentEditor editor, SwitchStatementSyntax switchStatement)
+        {
+            var analysis = editor.SemanticModel.AnalyzeDataFlow(switchStatement);
```

Comment:
You might also try just analyzing the specific case to try to deal with the scoping stuff we were discussing